packages/ripple-binary-codec/src/enums/README.md

```diff
@@ -53,8 +57,92 @@ See:
 - https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/TER.h
 - https://xrpl.org/transaction-results.html
 
-TODO: Write a script to read rippled's source file and generate the necessary mapping.
+To generate a new definitions file from rippled source code, use this tool: https://github.com/RichardAH/xrpl-codec-gen
 
 ## Transaction Types
 
 See https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/TxFormats.h
+
+# Defining Your Own Definitions
+
+If you're building your own sidechain or writing an amendment for the XRPL, you may need to create new XRPL definitions.
+
+To do that there are a couple things you need to do:
+
+1. Generate your own `definitions.json` file from rippled source code using [this tool](https://github.com/RichardAH/xrpl-codec-gen) (The default `definitions.json` for mainnet can be found [here](https://github.com/XRPLF/xrpl.js/blob/main/packages/ripple-binary-codec/src/enums/definitions.json))
+2. Create new SerializedType classes for any new Types (So that encode/decode behavior is defined)
+
+- For examples of how to implement that you can look at objects in the [`types` folder](../types/), such as `Amount`, `UInt8`, or `STArray`.
+
+3. Import your `definitions.json` file and `coreTypes` from the `types` folder, then use them to construct your own `XrplDefinitions` object.
+4. Pass the `XrplDefinitions` object whenever you `encode` or `decode` a transaction.
+5. If you added any new transaction types, you should create an `interface` for the transaction that extends `BaseTransaction` from the `xrpl` repo to use it with the functions on `Client` (See the below example of adding a new transaction type)
+
+## Example of adding a new Transaction type
+
+```
+// newTypeDefs is where you can import your custom defined definitions.json file
+const newTypeDefs = require('./new-type.json')
```

Comment:
The name `newTypeDefs` is a little confusing - it makes me thing that it's new types. Maybe it should be named something like `newDefinitions` instead.

Reply:
Renamed to `newDefinitionsJson` to avoid it conflicting too hard with `newDefs` which is the object we use throughout the code to replace the types (since `newDefinitions` and `newDefs` are basically the same meaning)

Reply:
Was this change pushed?

Reply:
Apparently not - updated :)

Reply:
Oh I see - I updated one snippet, but there were too - oops!